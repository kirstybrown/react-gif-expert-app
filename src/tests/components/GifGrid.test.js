import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Tests for GifGrid component', () => {

    const category = 'pandas';
  
    test('should show GifGrid component correctly', () => {
      
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow( <GifGrid category={category} />);
        expect( wrapper ).toMatchSnapshot();
    });

    test('should show items when images are loaded with useFetchGifs', () => {
      
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/anything.jpg',
            title: 'beautiful idea'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow( <GifGrid category={category} />);

        // expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists()).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

    });
    
    
});
