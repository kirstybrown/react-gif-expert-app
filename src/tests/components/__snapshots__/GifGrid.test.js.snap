// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifGrid component should show GifGrid component correctly 1`] = `
<Fragment>
  <h3
    className="category animate__animated animate__fadeIn"
  >
     
    pandas
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
